.github/workflows/commit.yaml

```diff
@@ -106,18 +106,22 @@ jobs:
         run: bash <(curl -s https://codecov.io/bash)
 
   test_scratch:
-    name: ${{ matrix.arch }}, Linux (scratch), Go-${{ matrix.go-version }}
-    runs-on: ubuntu-22.04
+    name: ${{ matrix.platform.arch }}, Linux (scratch), Go-${{ matrix.go-version }}
+    runs-on: ${{ matrix.platform.os }}
     strategy:
       fail-fast: false  # don't fail fast as sometimes failures are arch/OS specific
       matrix:  # Use versions consistent with wazero's Go support policy.
         go-version:
           - "1.23"  # Current Go version
           - "1.21"  # Floor Go version of wazero (current - 2)
-        arch:
-          - "amd64"
-          - "arm64"
-          - "riscv64"
+        platform:
+          - os: ubuntu-22.04
+            arch: amd64
+          - os: ubuntu-22.04-arm
+            arch: arm64
```

Comment:
Shouldn't we add to the normal `test` task too?

Another note is personally, running this for every arch seems like overkill. amd64 was added for #385, while arm64 was here because it required qemu. Not it doesn't require qemu, so I'm wondering if we need both amd64 and arm64 in this matrix.

Reply:
yeah I would rather delete this for amd64 and arm64

Reply:
I used this because it's a “reduced” test suite compared to “full” (same way I used it for the VMs, although cross compilation helps there).

But I can drop this and run the full test suite on Linux ARM.

Or not at all (we stopped running any tests on macOS AMD64). I run a bunch of these other configurations (including BSD ARM) on my own repo, so it's not like it's untested.

Reply:
so up to you but we want to reduce the CI costs

Reply:
Well at least for this run, scratch+AMD was faster than base+AMD (and scratch+ARM is faster still).

I dunno what's cheaper by the minute? 

If we wanna reduce costs we can drop some of the targets I added, or move them to something we manually run (e.g. before a release).

If we make them manually triggered I can at least run them for free on a fork, that's easy enough to do.